modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/DeleteDataStreamTransportAction.java

```diff
@@ -155,6 +155,9 @@ static ClusterState removeDataStream(
             DataStream dataStream = currentState.metadata().dataStreams().get(dataStreamName);
             assert dataStream != null;
             backingIndicesToRemove.addAll(dataStream.getIndices());
+            if (DataStream.isFailureStoreEnabled()) {
```

Comment:
This is actually one of those rare cases I think we don't want to check the flag. For example, if someone were to set the flag to true (via system properties), then play with it, then set it back to false, deleting the data stream would leave the failure store indices hanging. Do you think we need the flag here?

Reply:
I added it for consistency's sake. If you try this feature right now and then disable it afterward I'm pretty sure cluster state breaks to begin with since we conditionally execute the parsing code to pull this data based on the flag. I'd be ok with removing this flag check here because normal execution should see all this lists as empty. It makes me wonder if we should revisit how the cluster state parsing is handled.

Reply:
I went ahead and removed the flag, save us some keystrokes later.